### ISSUE TITLE
Kuutamo node does not keep producing block in unstable version `1.30.0-rc4`

### ISSUE THREAD

**Issue by yanganto** [kuutamolabs/near-staking-knd#321]
Currently, the validator node can not produce a block greater than `#12`(or a small number) with `neard` version `1.30.0-rc.4`, and version `1.29.0` works well.  

This issue might be a setting issue, not a bug.

- [x] #345
- [ ] #347

**Comment by yanganto** [comment]
This issue is not real in `main`, after kill `neard` of leader without kill `kuutamod` everything looks goos.
https://github.com/kuutamolabs/near-staking-knd/blob/29bdda2e3fc2de35f1d94e0c95b42516e368e86c/tests/test_multiple_nodes.py#L169-L171

**Comment by yanganto** [comment]
This issue happens in unstable not stable version, no matter killing `neard` or `kuutamod`.
https://github.com/kuutamolabs/near-staking-knd/pull/322/checks?check_run_id=9404696828

**Comment by yanganto** [comment]
This issue is not always reproducible.  Could we add a 10 seconds block-producing watcher when the state is just into `Voting` or `Validating`?  Once it happened we can easier to collect logs and also do some emergency treatment if any.  Besides, could we add a debug log feature on `kuutamod`, which will collect log around the state changing?

**Comment by yanganto** [comment]
This issue will not exist, if we implement key reload without restart.

As the log shows this happened after setting the validator because we restart the neard.
`2022-11-22T01:11:34.733527Z DEBUG handle_block_production: client: Cannot produce any block: not enough approvals beyond 12`

[error_debug.txt](https://github.com/kuutamolabs/near-staking-knd/files/10061901/error_debug.txt)

**Comment by yanganto** [comment]
```text
# bad: [7db3eac2daacf3bb00077fc21b56be1463c5b457] feat: switch to fetching state step on flat storage creation (#8019)
# good: [cdebd3fa2cf516f0b672710c404718c331dd1b07] 1.29.0
git bisect start '7db3eac2daacf3bb00077fc21b56be1463c5b457' 'cdebd3fa2cf516f0b672710c404718c331dd1b07'
# bad: [8bfcde874d96d998c7f063140fd08346cd0ee2a3] Revert "Update rocksdb to 0.19 (#7357)" (#7401)
git bisect bad 8bfcde874d96d998c7f063140fd08346cd0ee2a3
# good: [946f48ea0ab4f1259ef641572b0398558c784ebf] 1.29.0-rc.4
git bisect good 946f48ea0ab4f1259ef641572b0398558c784ebf
# bad: [62936b48617e5923792ec61916d99717ce9a39d5] docs: move epoch docs from confluence (#7838)
git bisect bad 62936b48617e5923792ec61916d99717ce9a39d5
# bad: [161e3e399b0e7d53dfa73aad7ddc943d3941c590] Simplify simple_nightshade shard layout config (#7499)
git bisect bad 161e3e399b0e7d53dfa73aad7ddc943d3941c590
# bad: [4f98a32e56cfdf60e182eda7d42cc54b0be2dee8] chore: upgrade anyhow and thiserror (#7463)
git bisect bad 4f98a32e56cfdf60e182eda7d42cc54b0be2dee8
# bad: [cfbb64558870d826fc25199c623b656481c595a3] reimplemented get_tier1_accounts to avoid expensive calls (#7402)
git bisect bad cfbb64558870d826fc25199c623b656481c595a3
# bad: [3b9012dd3f3cabd42a60107f497b9b72abf1acac] feat: trie cache configuration (#7578)
git bisect bad 3b9012dd3f3cabd42a60107f497b9b72abf1acac
# bad: [6e97cfde3d05e6d6e38dc9801d95a3ac38e90560] Do not forward the same chunk part twice. (#7512)
git bisect bad 6e97cfde3d05e6d6e38dc9801d95a3ac38e90560
# bad: [68bfa84ed1455f891032434d37ccad696e91e4f5] [SHRD-135] Don't route the messages through peers that are too far behind (#7322)
git bisect bad 68bfa84ed1455f891032434d37ccad696e91e4f5
# good: [d70b958470c695d4f632554374b8c0c7b49abed7] chore: bump version to 0.14.0 (#6918)
git bisect good d70b958470c695d4f632554374b8c0c7b49abed7
# bad: [fb747248112854bd8c3f450afa45d5630978308e] fix(uptime): duration formatting (#7328)
git bisect bad fb747248112854bd8c3f450afa45d5630978308e
# bad: [1897d5144a7068e4c0d5764d8c9180563db2fe43] Add RejectBlocksWithOutdatedProtocolVersions to the next shardnet release (#7538)
git bisect bad 1897d5144a7068e4c0d5764d8c9180563db2fe43
# good: [652ab15b24c2eeb6a39e487cf6a3623b0a055d8e] chore: bump version to 0.13.0 (#6830)
git bisect good 652ab15b24c2eeb6a39e487cf6a3623b0a055d8e
```

**Comment by yanganto** [comment]
Fixed by #351
